pkg/odo/genericclioptions/localprovider.go

```diff
@@ -35,7 +35,7 @@ func GetValidEnvInfo(cmdline cmdline.Cmdline) (*envinfo.EnvSpecificInfo, error)
 	if !envInfo.Exists() {
 		//revive:disable:error-strings This is a top-level error message displayed as is to the end user
 		return nil, errors.New(`The current directory does not represent an odo component.
-To start editing your component, use "odo dev" and open this folder in your favorite IDE. Changes will be directly reflected on the cluster.
+If you are working from an empty directory, use "odo init" to get started, else start editing your component, use "odo dev" and open this folder in your favorite IDE. Changes will be directly reflected on the cluster.
 To deploy your component to a cluster use "odo deploy".
 Or switch to directory with a component.`)
```

Comment:
How about rethinking this? I would go with bullets/list, and not go beyond 120 characters in a line. This one is a lot of content to read and make sense of.

Something like:

```
The current directory does not represent an odo component. 
* To get started with creating an odo component, create a new directory and cd into it.
* From that empty directory, use "odo init" to get started.
* Once you have initialized an odo component, you can start editing it in an IDE, and use `odo dev` to see your changes get reflected on the cluster.
```

I find mentioning `odo deploy` a bit pointless when the user is operating from an empty/incorrect directory. We might even the odo.dev docs if we want to be **SO** verbose!

Reply:
I like the idea about listing.

>I find mentioning odo deploy a bit pointless when the user is operating from an empty/incorrect directory.

I agree, I'll remove that.

>Let's even dump the entire odo.dev docs if we want to be SO verbose!

What do you think about pointing at odo.dev for more information then?